File: impl/app/android/app/src/main/java/com/ak/contexta/ui/components/AppTopBar.kt
package com.ak.contexta.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.RowScope
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.ArrowBack
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.ak.contexta.ui.theme.Background
import com.ak.contexta.ui.theme.MutedSoft

/** Unified app bar: 44dp back button + serif title + optional trailing actions. */
@Composable
fun AppTopBar(
    title: String,
    onBack: (() -> Unit)? = null,
    modifier: Modifier = Modifier,
    actions: @Composable RowScope.() -> Unit = {}
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .background(Background)
            .padding(horizontal = 12.dp, vertical = 6.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        if (onBack != null) {
            AppIconButton(
                icon = Icons.AutoMirrored.Outlined.ArrowBack,
                contentDescription = "返回",
                onClick = onBack,
                tint = MutedSoft
            )
            Spacer(modifier = Modifier.width(4.dp))
        }
        Text(
            text = title,
            style = MaterialTheme.typography.displayMedium,
            maxLines = 1,
            modifier = Modifier.weight(1f)
        )
        actions()
    }
}

/** Section label per prototype caption-upper: 12sp/500/+1.5sp uppercase. */
@Composable
fun SectionLabel(title: String, modifier: Modifier = Modifier) {
    Text(
        text = title.uppercase(),
        style = MaterialTheme.typography.labelSmall,
        color = MutedSoft,
        modifier = modifier.padding(top = 20.dp, bottom = 6.dp)
    )
}

File: impl/app/flutter/lib/core/components/app_top_bar.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';
import '../theme/app_dimens.dart';
import '../theme/app_type.dart';
import 'app_button.dart';

/// 统一顶栏：44dp 返回钮（MutedSoft）+ serif 标题 + 右侧 actions 槽位。
/// 对照 Kotlin ui/components/AppTopBar.kt。
class AppTopBar extends StatelessWidget {
  const AppTopBar({
    super.key,
    required this.title,
    this.onBack,
    this.actions = const [],
  });

  final String title;
  final VoidCallback? onBack;
  final List<Widget> actions;

  @override
  Widget build(BuildContext context) {
    return Container(
      color: AppColors.background,
      padding: const EdgeInsets.symmetric(
        horizontal: AppSpacing.sm,
        vertical: 6,
      ),
      child: Row(
        children: [
          if (onBack != null) ...[
            AppIconButton(
              icon: Icons.arrow_back,
              tooltip: '返回',
              onClick: onBack!,
              tint: AppColors.mutedSoft,
            ),
            const SizedBox(width: AppSpacing.xxs),
          ],
          Expanded(
            child: Text(
              title,
              style: AppType.textTheme.displayMedium,
              maxLines: 1,
              overflow: TextOverflow.ellipsis,
            ),
          ),
          ...actions,
        ],
      ),
    );
  }
}

/// 全大写分组标签：12sp/500/+1.5sp 字距。
/// 对照 Kotlin ui/components/AppTopBar.kt 的 SectionLabel。
class SectionLabel extends StatelessWidget {
  const SectionLabel(
    this.title, {
    super.key,
  });

  final String title;

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(top: 20, bottom: 6),
      child: Text(
        title.toUpperCase(),
        style: AppType.textTheme.labelSmall?.copyWith(
          color: AppColors.mutedSoft,
        ),
      ),
    );
  }
}
